src/main/java/com/internet/shop/controllers/AddProductController.java

```diff
@@ -0,0 +1,35 @@
+package com.internet.shop.controllers;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.math.BigDecimal;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/product/addProduct.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
```

Comment:
What happens if we receive the empty Strings?
Will you be able to work with them?

Reply:
Checklist says: "All checks to see if any field is empty or that field should accept only numbers should be done by jsp"
All fields on the jsp must be "required".